recipes/xz_utils/all/conanfile.py

```diff
@@ -28,159 +34,146 @@ class XZUtils(ConanFile):
             # https://developercommunity.visualstudio.com/content/problem/140294/windowstargetplatformversion-makes-it-impossible-t.html
             windows_target_platform_version_old = "<WindowsTargetPlatformVersion>10.0.15063.0</WindowsTargetPlatformVersion>"
-            if self.settings.compiler.version == 15:
+            if (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler) == "15") or \
+               (self.settings.compiler == "msvc" and Version(self.settings.compiler) == "191"):
                 windows_target_platform_version_new = "<WindowsTargetPlatformVersion>$([Microsoft.Build.Utilities.ToolLocationHelper]::GetLatestSDKTargetPlatformVersion('Windows', '10.0'))</WindowsTargetPlatformVersion>"
             else:
                 windows_target_platform_version_new = "<WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>"
-            tools.replace_in_file(os.path.join(self._source_subfolder, "windows", "vs2017", "liblzma.vcxproj"),
+            replace_in_file(self, os.path.join(self.source_folder, "windows", "vs2017", "liblzma.vcxproj"),
                                   windows_target_platform_version_old,
                                   windows_target_platform_version_new)
-            tools.replace_in_file(os.path.join(self._source_subfolder, "windows", "vs2017", "liblzma_dll.vcxproj"),
+            replace_in_file(self, os.path.join(self.source_folder, "windows", "vs2017", "liblzma_dll.vcxproj"),
                                   windows_target_platform_version_old,
                                   windows_target_platform_version_new)
 
-        # Allow to install relocatable shared lib on macOS
-        if tools.is_apple_os(self.settings.os):
-            tools.replace_in_file(
-                os.path.join(self._source_subfolder, "configure"),
-                "-install_name \\$rpath/",
-                "-install_name @rpath/",
-            )
-
     def _build_msvc(self):
         # windows\INSTALL-MSVC.txt
-        msvc_version = "vs2017" if Version(self.settings.compiler.version) >= "15" else "vs2013"
-        with tools.chdir(os.path.join(self._source_subfolder, "windows", msvc_version)):
+        if (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler) >= "15") or \
+           (self.settings.compiler == "msvc" and Version(self.settings.compiler) >= "191"):
+            msvc_version = "vs2017"
+        else:
+            msvc_version = "vs2013"
+        with chdir(os.path.join(self.source_folder, "windows", msvc_version)):
             target = "liblzma_dll" if self.options.shared else "liblzma"
             msbuild = MSBuild(self)
-            msbuild.build(
-                "xz_win.sln",
-                targets=[target],
-                build_type=self._effective_msbuild_type,
-                platforms={"x86": "Win32", "x86_64": "x64"},
-                upgrade_project=Version(self.settings.compiler.version) >= "17")
```

Comment:
I've lost this logic, I don't know what is the equivalent in the new `MSBuild` build helper.

Reply:
and it's required, it fails on Windows with Visual Studio 2022

Reply:
https://github.com/conan-io/conan/issues/12155&a